src/MessagePack/Internal/UnsafeRefSerializeHelper.cs

```diff
@@ -0,0 +1,1572 @@
+                        {
+                            loaded.AsByte().StoreUnsafe(ref Unsafe.Add(ref outputIterator, outputOffset));
+                            outputOffset += (nuint)Vector128<sbyte>.Count;
+                        }
+                        else
+                        {
+                            for (var i = 0; i < Vector128<sbyte>.Count; i++, lessThanMinFixNegativeInt >>= 1)
+                            {
+                                var inputValue = Unsafe.Add(ref input, inputOffset + (nuint)i);
+                                if ((lessThanMinFixNegativeInt & 1) == 1)
+                                {
+                                    Unsafe.Add(ref outputIterator, outputOffset) = MessagePackCode.Int8;
+                                    Unsafe.Add(ref outputIterator, outputOffset + 1) = (byte)inputValue;
+                                    outputOffset += 2;
+                                }
+                                else
+                                {
+                                    Unsafe.Add(ref outputIterator, outputOffset) = (byte)inputValue;
+                                    outputOffset += 1;
+                                }
+                            }
+                        }
+                    }
+
+                    writer.Advance((int)outputOffset);
+                    length -= (int)alignedInputLength;
+                    input = ref Unsafe.Add(ref input, alignedInputLength);
+                }
+            }
+        }
+#endif
+
+        while (length > 0)
+        {
+            var inputLength = length;
+            if (inputLength > maxInputSize)
+            {
+                inputLength = maxInputSize;
+            }
+
+            var destination = writer.GetSpan(inputLength * maxOutputElementSize);
+            ref var outputIterator = ref MemoryMarshal.GetReference(destination);
+            nuint outputOffset = 0;
+            for (nuint index = 0; index < (nuint)inputLength; index++)
+            {
+                writer.CancellationToken.ThrowIfCancellationRequested();
+                outputOffset += ReverseWriteUnknown(ref Unsafe.Add(ref outputIterator, outputOffset), Unsafe.Add(ref input, index));
+            }
+
+            writer.Advance((int)outputOffset);
+            length -= inputLength;
+            input = ref Unsafe.Add(ref input, inputLength);
+        }
+    }
+
+    internal static void Serialize(ref MessagePackWriter writer, ref short input, int length)
+    {
+        if (BitConverter.IsLittleEndian)
+        {
+            LittleEndianSerialize(ref writer, ref input, length);
+        }
+        else
+        {
+            BigEndianSerialize(ref writer, ref input, length);
+        }
+    }
+
+    private static void BigEndianSerialize(ref MessagePackWriter writer, ref short input, int length)
+    {
+        var i = 0;
+        do
+        {
+            writer.CancellationToken.ThrowIfCancellationRequested();
```

Comment:
I'm thinking this cancellation doesn't need to be _inside_ what is otherwise a fairly tight loop. One check before entering ought to be plenty IMO.

Reply:
Ok. I'll move this cancellation check to outside of this loop.